Create main.rb, this file use Bowling class and execute the game

jorge27 opened the pull request:
This file link all files and create the flow for the game.
In this case, here capture data input for call the class and later validate and calculate the punctuation.

jorge27 committed:
Created a tests for class bowling

The class bowling most have two methods, capture and calculate.
The method capture receive two values and validate if this values are
a numbers valids for append a punctuation
The method calculate, will calcule the punctuation of the shot

jorge27 committed:
Created class "bowling", the class for validate data and calcule points

This principal methods are two, the method capture validate bowling
data points, and method calculate, calcule the points of game. Also
contains method turn, this, return the turn of the game, and method rounds,
who return the bowling points of the turn indicate

jorge27 committed:
Correct identation, and changes some spec values

Correct identation, now are 2 spaces, change some spec values, now
instead of 0, spec nil

jorge27 committed:
Create main.rb, this file use Bowling class and execute the game

This file link all files and create the flow for the game.
In this case, here capture data input for call the class and later
validate and calculate the punctuation.

jorge27 committed:
Move file to app folder for use rubycritic

jorge27 committed:
Move file bowling.rb to folder app for execute rubycritic

jorge27 committed:
Modified main.rb, split the class Bowling in Bowling and Score

Modified main for append score class, and append the call to method
bowPrint, now, bowPrint is called from score class

ese-varo commented on app/lib/bowling.rb:
```diff
@@ -0,0 +1,109 @@
+    end
+    
+
+    if x.negative? || y.negative? || x > 10 || x > 10
+      return nil
+    end
```
podrias reducirlo a una linea: 
```suggestion
  return nil if x.negative? || y.negative? || x > 10 || x > 10
```
lo mismo puedes hacer en varios de tus if

ese-varo commented on app/lib/bowling.rb:
```diff
@@ -0,0 +1,109 @@
+    for i in 0..130
+      print "-"
+    end
+    print "\n"
+    for i in 0..9
+      print "|"
+      for j in 0..10
+        print " "
+      end
+      print "|"
+    end
+    print "\n"
+    for i in 1..10
+      print "|    %3s    |" % [@points[i-1]]
+    end
+    print "\n"
+    for i in 0..9
+      print "|"
+      for j in 0..10
+        print " "
+      end
+      print "|"
+    end
+    print "\n"
```
puedes hacer una funcion para ejecutar esos for anidados, asi no tendrias que repetirlo dos veces, puedes crear un metodo privado para ya que solo lo usas dentro de la clase: 

```suggestion
    your_method
    for i in 1..10
      print "|    %3s    |" % [@points[i-1]]
    end
    your_method
```

ese-varo commented on app/lib/bowling.rb:
```diff
@@ -0,0 +1,109 @@
+    end
+
+    if (@turn!=10) && ((x + y).negative? || (x + y) > 10)
+      return nil
```
el `return` en la ultima linea de tu codigo (en este caso es la primera y la ultima) no es necesario que lo escribas ya que es implicito ruby regresa la evaluacion de la ultima expresion que encuentra en tu bloque

ese-varo commented on app/lib/bowling.rb:
```diff
@@ -0,0 +1,109 @@
+    @points
+  end
+
+  def bowPrint
```
los metodos se nombran utilizando snake_case, por ejemplo
```suggestion
  def bow_print
```

jorge27 committed:
Modified print score method, now for print board call score.bow_print

Now, for call the method who print the score board used bow_print in
class score
